orangecontrib/text/widgets/ownyt.py

```diff
@@ -173,67 +163,100 @@ def select_query(self, n):
-
-                # Enable 'retrieve remaining' button.
-                if self.num_retrieved < min(self.all_hits, 1000):
-                    self.retrieve_other_button.setText('Retrieve remaining records ({})'
-                                                       .format(min(self.all_hits, 1000)-self.num_retrieved))
-                    self.retrieve_other_button.setEnabled(True)
-                    self.retrieve_other_button.setFocus()
-                else:
-                    self.retrieve_other_button.setText('All records retrieved')
-                    self.retrieve_other_button.setEnabled(False)
-
-                # Add the query to history.
-                if qkw not in self.recent_queries:
-                    self.recent_queries.insert(0, qkw)
-            else:
-                if error:
-                    if isinstance(error, HTTPError):
-                        self.error(1, "An error occurred(HTTP {})".format(error.code))
-                    elif isinstance(error, URLError):
-                        self.error(1, "An error occurred(URL {})".format(error.reason))
+        if not self.nyt_api:
+            return
+
+        # Interrupts on faulty inputs. #
+        # Query keywords.
+        qkw = self.query_combo.currentText()
+        if not qkw:
+            self.warning(1, "Please enter a query before attempting to fetch results.")
+            return
+
+        # Text fields.
+        text_includes_params = [self.includes_headline, self.includes_lead_paragraph, self.includes_snippet,
+                                self.includes_abstract, self.includes_keywords]
+        if True not in text_includes_params:
+            self.warning(1, "You must select at least one text field.")
+            return
+
+        # Year span.
+        if self.year_from and not self.year_from.isdigit() or \
+                self.year_to and not self.year_to.isdigit():
+            self.warning(1, "The time interval endpoints must be digits.")
+            return
+
+        # Warnings on bad inputs. #
+        if self.year_from and self.year_from > self.year_to:
+            self.warning(1, "The end time is greater than the starting time.")
+
+        if self.year_from and int(self.year_from) < 1851:
+            self.warning(1, "There are no records before the year 1851. "
+                            "Assumed 1851 as start date for this query.")
+
+        # Set the query url.
+        required_text_fields = [tp for tf, tp in zip(text_includes_params, self.text_fields) if tf]
+        self.nyt_api.set_query_url(qkw, self.year_from,
+                                   self.year_to,
+                                   required_text_fields)
+
+        # Execute the query.
+        res, cached, error = self.nyt_api.execute_query(0)
+
+        if not res:     # Just return if no response.
+            if error:
+                self.display_error_response(error)
+            return
+
+        # Construct a corpus for the output.
```

Comment:
The creation of the Corpus from the json should also be moved to scripting part (approx lines 224–237). Think about how would you now generate a Corpus if using the scripting part of the addon?